### DISCUSSION TITLE
Did you ever manage to start an interactive getty on the console?

### DISCUSSION THREAD

**Discussion by deitch** [Code-Hex/vz#26] [Q&A]
Did you succeed? What device did you use? It looks, based on your description, that it should be `/dev/hvc0`, but that doesn't seem to do anything.

**Comment by cfergeau** [comment]
I can interact with the VM terminal after starting it with example/virtualization if this is what you are asking?

**Comment by deitch** [comment]
Hmm. Then I'm wondering what I'm doing wrong. 

I'm running it not with a standard distro, but with a composed minimal one. It's got a small script to read cmdline and determine where to start getty. Yet despite starting several - including to /dev/hvc0 - nothing is interacting. 

Do you know what device the actual console is using?

**Reply by deitch** [reply]
It turns out to be using `hvc0`, as we suspected; I had permissions wrong on my getty in the minimal OS. This part is fine now.

**Comment by deitch** [comment]
Actually, I tried your approach. I just built from `example/` dir, pulled down the Ubuntu 20.04.3 iso, extracted the vmlinuz and initrd, set them to env vars, and tried to boot it. It just sits there.

**Comment by cfergeau** [comment]
You must gunzip the kernel or it will just sit there indeed. (forgot if this step is m1 specific, or is also needed on x86_64)

**Comment by deitch** [comment]
It is an arm64 issue. For some reason, arm64 kernels are gzipped (which qemu and others can handle, but macOS VF cannot 🤷‍♂️ ) while x86_64 use built-in compression.

It isn't that, I definitely was using an extracted kernel.

**Reply by deitch** [reply]
The uncompressed kernel requirement is [well-documented at kernel.org](https://www.kernel.org/doc/Documentation/arm64/booting.txt). Apparently qemu can handle this, because qemu unzips the kernel in memory, and then passes that memory pointer to the VM.

Since it is an `NSURL` under the cover, using a FileURL [here](https://github.com/Code-Hex/vz/blob/6e4da14d1308e6af339f76eb92b7ac4269c6ac73/virtualization.m#L50-L59), we could use a `URLWithString` with base64-encoded data.

Would you consider an option to pass a `[]byte` to [NewLinuxBootLoader](https://github.com/Code-Hex/vz/blob/master/bootloader.go#L70-L89) ? Or at least a base64-encoded string?

**Reply by deitch** [reply]
Proposed a PR, see #27
